src/librbd/mirror/snapshot/PromoteRequest.h

```diff
@@ -28,23 +28,24 @@ class PromoteRequest {
   static PromoteRequest *create(ImageCtxT *image_ctx,
                                 const std::string& global_image_id,
                                 const std::string &group_snap_id,
-                                uint64_t *snap_id, Context *on_finish) {
+                                uint64_t rollback_snap_id, uint64_t *snap_id,
+                                Context *on_finish) {
     return new PromoteRequest(image_ctx, global_image_id, group_snap_id,
-                              snap_id, on_finish);
+                              rollback_snap_id, snap_id, on_finish);
   }
   static PromoteRequest *create(ImageCtxT *image_ctx,
                                 const std::string& global_image_id,
                                 Context *on_finish) {
-    return new PromoteRequest(image_ctx, global_image_id, {}, nullptr,
+    return new PromoteRequest(image_ctx, global_image_id, {}, false, nullptr,
```

Comment:
Why is this passing false for a uint64_t rollback_snap_id?

Reply:
Ack, missed it. changed it to CEPH_NOSNAP